src/main/java/org/wildfly/security/auth/server/event/RealmIdentityCredentialUpdateEvent.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2016 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.security.auth.server.event;
+
+import org.wildfly.security.auth.server.RealmIdentity;
+import org.wildfly.security.auth.server.RealmUnavailableException;
+import org.wildfly.security.credential.Credential;
+
+/**
+ * An event indicating a credential change for a realm identity.
+ *
+ * @author <a href="mailto:fjuma@redhat.com">Farah Juma</a>
+ */
+public final class RealmIdentityCredentialUpdateEvent extends RealmEvent {
```

Comment:
The purpose of this class appears to be to replace the mechanism of the "credential" field on RealmSuccessfulAuthenticationEvent.

I think if we're going to go this route, we probably should instead simply forego the usage of events for this and have the callback handler either directly modify the credential for the realm, or else have a special/separate method on ModifiableSecurityRealm or ModifiableRealmIdentity (or even just on RealmIdentity) which is called when a credential update occurs.  This method could reject the update if the realm does not support it.  If a method on RealmIdentity is used, the default could always reject.  The default method on the ModifiableRealmIdentity class could update the credential the "usual" way (though we as of yet do not have a credential query method it appears).

This would avoid moving to a system of identity events that are pupose-driven (mirroring callbacks, in a way, to effect specific changes on the identity, where the realm reacts in an API-defined way) versus the existing system that are informative (to update the realm on the events surrounding the selected identity and allow the realm to react in a realm-policy-driven way).  WDYT?